src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -117,21 +119,28 @@ build_mhd_response(const struct sol_http_response *response)
 
     SOL_HTTP_PARAM_FOREACH_IDX (&response->param, value, idx) {
         int ret;
-        char buffer[512];
+        char buffer[512], key[512], value_str[512];
 
         switch (value->type) {
         case SOL_HTTP_PARAM_HEADER:
-            if (MHD_add_response_header(r, value->value.key_value.key, value->value.key_value.value) == MHD_NO) {
-                SOL_WRN("Could not add the header: %s", value->value.key_value.key);
+            ret = snprintf(key, sizeof(key), "%.*s", SOL_STR_SLICE_PRINT(value->value.key_value.key));
+            if (ret < 0 || ret >= (int)sizeof(key))
+                goto err;
+            ret = snprintf(value_str, sizeof(value_str), "%.*s", SOL_STR_SLICE_PRINT(value->value.key_value.value));
+            if (ret < 0 || ret >= (int)sizeof(value_str))
+                goto err;
+            if (MHD_add_response_header(r, key, value_str) == MHD_NO) {
+                SOL_WRN("Could not add the header: %.*s", SOL_STR_SLICE_PRINT(value->value.key_value.key));
                 goto err;
             }
             break;
         case SOL_HTTP_PARAM_COOKIE:
-            ret = snprintf(buffer, sizeof(buffer), "%s=%s", value->value.key_value.key, value->value.key_value.value);
+            ret = snprintf(buffer, sizeof(buffer), "%.*s=%.*s", SOL_STR_SLICE_PRINT(value->value.key_value.key),
```

Comment:
is `value->value.key_value.key` and value->value.key_value.value` encoded? If not, then we have problems.

if we move to sol_buffer, we can also offer a way to encode to a buffer, making it even simpler to use.